warrior/Framework/Utils/cli_Utils.py

```diff
@@ -662,25 +663,76 @@ def _send_cmd(obj_session, **kwargs):
 
 def _get_response_dict(details_dict, index, response, response_dict):
     """Get the response dict for a command. """
+    def print_warn_msg(keyvars, numpats):
+        """ print a warning message if the number of vars to be stored with
+        the patterns does not match with the number of patterns
+        """
+        warn_msg = ("The number of response reference keys to store is {0} than \nthe response "
+                    "reference patterns.\nThe number of response reference keys({1}) is {2}\n"
+                    "which is {0} than number of response reference patterns {3}")
+        lessormore = "less" if len(keyvars) < numpats else "more"
+        print_warning(warn_msg.format(lessormore, ", ".join(keyvars), len(keyvars), numpats))
     resp_ref = details_dict["resp_ref_list"][index]
     resp_req = details_dict["resp_req_list"][index]
     resp_pat_req = details_dict["resp_pat_req_list"][index]
+    resp_keys = details_dict["resp_key_list"][index]
+    inorder = details_dict["inorder_resp_ref_list"][index]
+    status = True
+    if inorder is not None and inorder.lower().startswith("n"):
+        inorder = False
+    else:
+        inorder = True
 
     resp_req = {None:'y', '':'y',
                 'no':'n', 'n':'n'}.get(str(resp_req).lower(), 'y')
     resp_ref = {None:index+1, '':index+1 }.get(resp_ref, str(resp_ref))
     #response_key=resp_ref_list[i] if resp_ref_list[i] else i+1
     if not resp_req=="n":
+        save_msg1 = "User has requested saving response"
+        save_msg2 = "Response pattern required by user is : {0}"
+        save_msg3 = "Portion of response saved to the data repository with key: {0}, value: {1}"
         if resp_pat_req is not None:
             # if the requested pattern not found return empty string
             reobj=re.search(resp_pat_req, response)
             response=reobj.group(0) if reobj is not None else ""
-            pNote("User has requested saving response. Response pattern required by user is : {0}".format(resp_pat_req))
-            pNote("Portion of response saved to the data repository with key: {0}, value: {1}".format(resp_ref, response))
+            response_dict[resp_ref] = response
+            pNote(save_msg1+'.')
+            pNote(save_msg2.format(resp_pat_req))
+            pNote(save_msg3.format(resp_ref, response))
+        elif resp_keys is not None:
+            keys = resp_ref.split(',')
+            patterns = [k.get("resp_pattern_req") for k in resp_keys]
+            if len(keys) != len(patterns):
+                print_warn_msg(keys, len(patterns))
+            if inorder:
+                pNote(save_msg1+' inorder.')
+                cpatterns = map(lambda s: "({})".format(s), patterns)
+                pattern = ".*".join(cpatterns)
```

Comment:
In-order pattern search is failing when the key values are from different lines.

Log:
-I- Response:
**ls NonExistTitleRowDir
ls: cannot access NonExistTitleRowDir: No such file or directory
vsethura@cinplgn04%**

-D- [2017-09-29 11:27:08] Command completed successfully
-I- Command Duration: 1.0 sec
-D- Verification required for command: 'ls NonExistTitleRowDir'
-D- Found  'ls NonExistTitleRowDir' in  response to 'ls NonExistTitleRowDir' on rtxttalp01_sys_1 :[pattern matched]:
-I- COMMAND STATUS:PASS
-I- User has requested saving response inorder.
-E- inorder search of patterns in response failed
-E- Expected: '(ls NonExistTitleRowDir).*(cannot access.*:).*(No such file or directory)'
-E- But Found: 'ls NonExistTitleRowDir
ls: cannot access NonExistTitleRowDir: No such file or directory

Reply:
@navvrnd Seems this comment is not adressed